### PR TITLE
Define and apply the project style

### PR DESCRIPTION
**Pull request by abrauchli**
Define and use the default project style with `clang-format`

### CODE REVIEW

File: sps30-uart/sps30.c [hunk 1]
```diff
@@ -18,36 +18,38 @@
...
 #include "sensirion_shdlc.h"
 #include "sps_git_version.h"
-#include "sps30.h"
 
 #define SPS_ADDR 0x00
 #define SPS_CMD_START_MEASUREMENT 0x00
 #define SPS_CMD_STOP_MEASUREMENT 0x01
-#define SPS_SUBCMD_MEASUREMENT_START {0x01, 0x03}
+#define SPS_SUBCMD_MEASUREMENT_START                                           \
+    { 0x01, 0x03 }
```
Comment (rnestler) [inline sps30-uart/sps30.c]: These look a bit confusing to me.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+Language: Cpp
```
Comment (rnestler) [inline .clang-format]: We actually use C, but I'm not sure what the implications would be.

Comment (rnestler) [reply .clang-format]: Never mind:
> LK_Cpp (in configuration: Cpp) Should be used for C, C++.
> *https://clang.llvm.org/docs/ClangFormatStyleOptions.html*

File: sps30-uart/sps30.c (hunk 1 above)
Comment (abrauchli) [reply sps30-uart/sps30.c]: Agreed, I saw those too but there doesn't seem to be control over that, short of `// clang-format off .. // clang-format on` hacks, so I figured I could live with it.

I played a bit with an online interactive configurator and also all provided default styles have it wrapped on the next line.
https://zed0.co.uk/clang-format-configurator/

**Pull request merged**